Makefile

```diff
@@ -261,8 +261,11 @@ examples:
 # api machinery regenerate
 
 apimachinery:
-	#go install ./cmd/libs/go2idl/conversion-gen
+	# conversion: go install ./cmd/libs/go2idl/conversion-gen
 	${GOPATH}/bin/conversion-gen --skip-unsafe=true --input-dirs k8s.io/kops/pkg/apis/kops/v1alpha1 --v=8  --output-file-base=zz_generated.conversion
+	# defaulters: go install ./cmd/libs/go2idl/defaulter-gen
+	${GOPATH}/bin/defaulter-gen --input-dirs k8s.io/kops/pkg/apis/kops/v1alpha1 --v=8  --output-file-base=zz_generated.defaults
+	${GOPATH}/bin/defaulter-gen --input-dirs k8s.io/kops/pkg/apis/kops/v1alpha2 --v=8  --output-file-base=zz_generated.defaults
```

Comment:
This generates the defaulter functions.  I hadn't wired them up before, but this is where defaulting _should_ be done for API objects.  Hopefully we can move lots of mechanical defaulting into here.  If you're adding a required field, it should be defaulted here.  Some cases are less simple - for example CIDR allocation probably _doesn't_ belong here.

Validation is the next big piece of the API to wire up, hopefully replacing a lot of our ad-hoc stuff.

Also, they are different targets - one is v1alpha1, the other is v1alpha2